### ISSUE TITLE
closing one-off task causes panic: close of nil channel

### ISSUE THREAD

**Issue by pieterjanpintens** [xelalexv/dregsy#35]
hi, i'm using your tool to sync a bunch of docker images from dockerhub to ecr.
It worked fine till a certain point where it get this `panic: close of nil channel`

The following is part of the log.

```
... a whole bunch of successful syncs ...
08:22:40  time="2020-12-11T07:22:40Z" level=debug msg="Copying config sha256:38983bba610e04c6bd00ef5c26b86ad5cbff75cb6b83431595d6fca744a6f135"
08:22:40  time="2020-12-11T07:22:40Z" level=debug msg="Writing manifest to image destination"
08:22:40  time="2020-12-11T07:22:40Z" level=debug msg="Storing signatures"
08:22:40  time="2020-12-11T07:22:40Z" level=debug msg="stopping tasks" func="github.com/xelalexv/dregsy/internal/pkg/sync.(*Sync).SyncFromConfig" file="/go/src/dregsy/internal/pkg/sync/sync.go:149"
08:22:40  panic: close of nil channel
08:22:40  
08:22:40  goroutine 1 [running]:
08:22:40  github.com/xelalexv/dregsy/internal/pkg/sync.(*Task).stopTicking(0xc0000bdf00)
08:22:40  	/go/src/dregsy/internal/pkg/sync/task.go:138 +0x40
08:22:40  github.com/xelalexv/dregsy/internal/pkg/sync.(*Sync).SyncFromConfig(0xc0003a02e0, 0xc0001d4ea0, 0x0, 0x0)
08:22:40  	/go/src/dregsy/internal/pkg/sync/sync.go:152 +0x59b
08:22:40  main.main()
08:22:40  	/go/src/dregsy/cmd/dregsy/main.go:113 +0x24f
```

This is my config file

```
relay: skopeo
tasks:
  - name: task1
    verbose: true
    source:
      registry: docker.io
    target:
      registry: 754813824532.dkr.ecr.eu-west-1.amazonaws.com
      auth-refresh: 10h
    mappings:
      - from: alpine
        tags: ['latest']
      - from: postman/newman
        tags: ['latest','5-alpine']
      - from: aequitas/http-api-resource
        tags: ['latest']
      - from: python
        tags: ['2.7.13-alpine','3-alpine']
      - from: node
        tags: ['10-alpine']
      - from: skyscrapers/apt-package-resource
        tags: ['latest']
      - from: maven
        tags: ['3.6.3-jdk-8','3.5.2-jdk-8','3.6.3-jdk-11']
      - from: openapitools/openapi-generator-cli
        tags: ['v4.3.1']
      - from: selenium/standalone-chrome
        tags: ['3.141.59-20200719']
      - from: mysql
        tags: ['5.7']
      - from: pjpinten/packer-g10k
        tags: ['1.2']
      - from: renovate/renovate
        tags: ['latest']
```
It is ran as a jenkins job, so i do the scheduling there. This is a one shot job.
Jenkins executes:

docker run --rm -v /opt/jenkins/workspace/ose-Concourse_sync-docker_master/config.yaml:/config.yaml -e LOG_METHODS=true -e LOG_LEVEL=trace -e AWS_ACCESS_KEY_ID=**** -e AWS_SECRET_ACCESS_KEY=**** xelalex/dregsy

Running this on my local machine works fines so maybe there is some kind of weird race condition?
I'm not to familiar with go. But it seems that there are timers (threads) in use which might cause this.
For me this is not really blocking as the sync itself works fine but the command does fail.
This is not very clean and my team asks questions about it.

**Comment by xelalexv** [comment]
It's not a race condition, but actually a hard error that occurs with one-off tasks. I recently changed the code for stopping a task, with the intention to make the process cleaner :rofl: The changes work with periodic task, but I did not test with one-off tasks. I added a test case and can reproduce 100%. I created a fix in branch `issue35` and pushed an according Docker image `xelalex/dregsy:issue35`. Could you validate with this? If all is ok, I'll create a new release.

**Comment by xelalexv** [comment]
Any feedback? Otherwise I'll just release this, tests are looking fine.

**Comment by pieterjanpintens** [comment]
Sure release it, i will test it

**Comment by xelalexv** [comment]
fixed with PR #36
